src/test/java/org/jenkinsci/plugins/enhancedoldbuilddiscarder/EnhancedOldBuildDiscarderTest.java

```diff
@@ -106,42 +81,44 @@ public void testPerformHoldMaxBuildsSecondCnd() throws Exception {
 		// while remaining build history is cleared since it exceeds max age
 		FreeStyleProject p = projectInstantiation();
 
-		EnhancedOldBuildDiscarder publisher = (new EnhancedOldBuildDiscarder(
+		EnhancedOldBuildDiscarder publisher = spy(new EnhancedOldBuildDiscarder(
 				"10", "5", "", "",
 				false, true));
 
+		when(publisher.ageUnitTest()).thenReturn(true);
 		publisher.perform(p);
-		List<? extends Run<?,?>> buildListPost = p.getBuilds();
-		assertEquals(5, buildListPost.size());
+		List<? extends Run<?,?>> buildList = p.getBuilds();
+		assertEquals(5, buildList.size());
 	}
 
 	@Test
 	public void testPerformHoldMaxBuildsThirdCnd() throws Exception {
 		// testing for circumstance where no builds are cleared since logs are beneath max age
-		// despite exceeding max build quantity
+		// despite exceeding max build quantity, for this reason ageUnitTest() is not stubbed
 		FreeStyleProject p = projectInstantiation();
 
-		EnhancedOldBuildDiscarder publisher = (new EnhancedOldBuildDiscarder(
-				"100000", "5", "", "",
+		EnhancedOldBuildDiscarder publisher = spy(new EnhancedOldBuildDiscarder(
+				"10", "5", "", "",
 				false, true));
 
 		publisher.perform(p);
-		List<? extends Run<?,?>> buildListPost = p.getBuilds();
-		assertEquals(10, buildListPost.size());
+		List<? extends Run<?,?>> buildList = p.getBuilds();
+		assertEquals(10, buildList.size());
 	}
 
 	@Test
 	public void testPerformHoldMaxBuildsFourthCnd() throws Exception {
-		// testing for circumstance where max builds is set to 0 and days to keep is not
-		// forcing deletion of all builds exceeding age, with no effect from max build quantity
+		// testing for circumstance where max builds is 1 and days to keep is 1
+		// forcing deletion of all builds except the last successful one
 		FreeStyleProject p = projectInstantiation();
 
-		EnhancedOldBuildDiscarder publisher = (new EnhancedOldBuildDiscarder(
-				"10", "", "", "",
+		EnhancedOldBuildDiscarder publisher = spy(new EnhancedOldBuildDiscarder(
```

Comment:
Btw, if you're not actually using any of the mockito stuff, you can just inline subclass it

new EnhancedOldBuildDiscarder("1", "1", "", "", false, true)) { 
@Override
protected void ageUnitTest() { return true; }
}

Not sure if thats easier/more readable though

Reply:
I'd be just as inclined to keep Mockito for this for legibility, I did remove the spy() from testPerformHoldMaxBuildsThirdCnd() though, since I don't stub for that one.